test/gateway/pollingInterval.js

```diff
@@ -488,3 +489,302 @@ test('Polling schemas (cache should be cleared)', async (t) => {
   await userService.close()
   clock.uninstall()
 })
+
+test('Polling schemas (subscriptions should be handled)', async (t) => {
+  const clock = FakeTimers.install({
+    shouldAdvanceTime: true,
+    advanceTimeDelta: 40
+  })
+
+  const user = {
+    id: 'u1',
+    name: 'John',
+    lastName: 'Doe'
+  }
+
+  const resolvers = {
+    Query: {
+      me: (root, args, context, info) => user
+    },
+    Mutation: {
+      triggerUser: async (root, args, { pubsub }) => {
+        await pubsub.publish({
+          topic: 'UPDATED.USER',
+          payload: {
+            updatedUser: user
+          }
+        })
+
+        return true
+      }
+    },
+    Subscription: {
+      updatedUser: {
+        subscribe: async (root, args, { pubsub }) =>
+          pubsub.subscribe('UPDATED.USER')
+      }
+    },
+    User: {
+      __resolveReference: (user, args, context, info) => user
+    }
+  }
+
+  const userService = Fastify()
+  const gateway = Fastify()
+
+  t.tearDown(() => {
+    userService.close()
+    gateway.close()
+  })
+
+  userService.register(GQL, {
+    schema: `
+      extend type Query {
+        me: User
+      }
+
+      extend type Subscription {
+        updatedUser: User
+      }
+
+      extend type Mutation {
+        triggerUser: Boolean
+      }
+
+      type User @key(fields: "id") {
+        id: ID!
+        name: String!
+      }
+    `,
+    resolvers: resolvers,
+    federationMetadata: true,
+    subscription: true
+  })
+
+  await userService.listen(0)
+
+  const userServicePort = userService.server.address().port
+
+  gateway.register(GQL, {
+    subscription: true,
+    gateway: {
+      services: [
+        {
+          name: 'user',
+          url: `http://localhost:${userServicePort}/graphql`,
+          wsUrl: `ws://localhost:${userServicePort}/graphql`
+        }
+      ],
+      pollingInterval: 2000
+    }
+  })
+
+  await gateway.listen(0)
+
+  const ws = new WebSocket(
+    `ws://localhost:${gateway.server.address().port}/graphql`,
+    'graphql-ws'
+  )
+
+  const client = WebSocket.createWebSocketStream(ws, {
+    encoding: 'utf8',
+    objectMode: true
+  })
+
+  t.tearDown(() => {
+    client.destroy()
+  })
+
+  client.setEncoding('utf8')
+
+  client.write(
+    JSON.stringify({
+      type: 'connection_init'
+    })
+  )
+
+  client.write(
+    JSON.stringify({
+      id: 1,
+      type: 'start',
+      payload: {
+        query: `
+          subscription {
+            updatedUser {
+              id
+              name
+            }
+          }
+        `
+      }
+    })
+  )
+
+  const promise = () =>
+    new Promise((resolve) => {
```

Comment:
please avoid the promise constructor. You can use `require('events').once()` to achieve the same result.

Reply:
Thank you. I replaced all promise constructors with `.once()`